### ISSUE TITLE
[Feature Request]: Adding icon support for small size badge

### ISSUE THREAD

**Issue by mehmetranas** [Trendyol/baklava#669]
### Describe the problem or use case

Currently we are utilizing the medium size badge, but we noticed that it doesn't fit well with our UI. Therefore, we would like to switch to the small size badge. However, the small size badge does not support icons.

### Proposed solution

Add icon support for small size badge. The icon may be 12x12 pixels.

**Comment by AykutSarac** [comment]
I'm guessing it might be because the icon becomes too small to view but @buseselvi might have better insights on it.

**Comment by buseselvi** [comment]
Yes, we didn't add it because the size is too small and we didn't need it until now. But if we needed to use it now we can add it, just make sure that the icon we use in the small badge is visible on the screen. Here is the [design link](https://www.figma.com/file/RrcLH0mWpIUy4vwuTlDeKN/Baklava-Design-Guide?type=design&node-id=10195%3A19559&mode=design&t=JoTFujqGAL5y153X-1) for adding that icon. 🚀
